### PR TITLE
Added getSocketDescriptor() to http::server

### PR DESCRIPTION
**Pull request by rdumusc**

### CODE REVIEW

**Comment by eile** [comment]
+1

File: zeq/http/server.cpp [hunk 1]
```diff
@@ -310,6 +310,18 @@ const servus::URI& Server::getURI() const
     return _impl->uri.toServusURI();
 }
 
+SocketDescriptor Server::getSocketDescriptor() const
```
Comment (bilgili) [inline zeq/http/server.cpp]: Returning a low-level fd from the Server seems fishy to me.

Comment (rdumusc) [inline zeq/http/server.cpp]: Having access to the socket handle is very useful in conjunction with a QSocketNotifier. It handles the poll() as part of the QEventLoop and emits an activated() QSignal when the socket has new data available. This signal can be connected to a lamba which calls recieive(), so with a few lines of code you have a server which automatically receives without the need for a separate processing thread. I see no easy alternative to that mechanism.

Comment (bilgili) [inline zeq/http/server.cpp]: I totally understand it. But still looks fishy :)

Comment (eile) [inline zeq/http/server.cpp]: It's a commonly used pattern. You get an opaque handle to poll, but should not use it for something else. ZMQ provides it, X11 ConnectionNumber does, and many more.

Comment (bilgili) [inline zeq/http/server.cpp]: I did not disagree, but it feels like there should be extendable layer around it. A socket and poll classes. In C world it is more understandable but in C++ world these can be wrapped nicer.

**Pull request merged**